client/asset/eth/eth.go

```diff
@@ -1459,24 +1465,35 @@ func (w *assetWallet) Redeem(form *asset.RedeemForm, feeWallet *assetWallet) ([]
 		candidateLimit := gasEst * 11 / 10 // Add 10% for good measure.
 		// Cannot be more than double.
 		if candidateLimit > gasLimit*2 {
-			return nil, nil, 0, fmt.Errorf("cannot recover from excessive gas estimate %d > 2 * %d", candidateLimit, gasLimit)
+			return fail(fmt.Errorf("cannot recover from excessive gas estimate %d > 2 * %d", candidateLimit, gasLimit))
```

Comment:
What was this double rule about?

Reply:
If the estimate for the amount of gas the redemption will use is more than double the hardcoded value, then it will fail.

Reply:
Fail where? Why?

Reply:
It won't submit the redemption transaction. The redemption function just returns an error. This is the current logic.

Reply:
This shouldn't happen though since there's only one code path in the contract.

Reply:
I meant that this function would fail. It is completely arbitrary for sure.. do you think it should be higher than 2x?

Reply:
Ohhhh.  Ok that's fine.  The comments suggested to me that there's something special about this threshold.  It's fine.  Could be less even, let's just keep it this way though.